### PR TITLE
[FLINK-14948][client] Implement shutDownCluster for MiniClusterClient

### PR DESCRIPTION
**Pull request by TisonKun**
## What is the purpose of the change

Indeed, we *can* implement proper `shutDownCluster` method for `MiniClusterClient`

## Verifying this change

Straightforward.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (not applicable)

cc @kl0u @aljoscha @zjffdu

### CODE REVIEW

File: flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java [hunk 1]
```diff
@@ -65,9 +65,7 @@ default void close() throws Exception {
 	/**
 	 * Shut down the cluster that this client communicate with.
 	 */
-	default void shutDownCluster() {
-		throw new UnsupportedOperationException();
-	}
+	void shutDownCluster();
```
Comment (zjffdu) [inline flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java]: What about classes that extend ClusterClient ?

Comment (TisonKun) [reply flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java]: Within Flink scope we implement this method for all cluster client and I think the consensus is that `ClusterClient` is an internal concept at least for now. Besides, every implementation of `ClusterClient` should think of whether or not and how to shut down its associated cluster.

However, although it makes no sense add a default implementation just `throw new UnsupportedOperationException()`, if you insist I can add back such default implementation just for outside implementation pass compiler checking.

Comment (zjffdu) [reply flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java]: Oh, my misunderstanding. I took it as new method, and thought other classes has no implementation for this method.

**Comment by TisonKun** [comment]
Thanks for your review! Merged.

**Pull request merged**